packages/flutter_tools/test/general.shard/build_system/targets/dart_test.dart

```diff
@@ -46,434 +42,443 @@ void main() {
           kTargetPlatform: getNameForTargetPlatform(TargetPlatform.android_arm),
         },
       );
+      androidEnvironment.buildDir.createSync(recursive: true);
       iosEnvironment = Environment.test(
         globals.fs.currentDirectory,
         defines: <String, String>{
           kBuildMode: getNameForBuildMode(BuildMode.profile),
           kTargetPlatform: getNameForTargetPlatform(TargetPlatform.ios),
         },
       );
-      HostPlatform hostPlatform;
-      if (globals.platform.isWindows) {
-        hostPlatform = HostPlatform.windows_x64;
-      } else if (globals.platform.isLinux) {
-        hostPlatform = HostPlatform.linux_x64;
-      } else if (globals.platform.isMacOS) {
-        hostPlatform = HostPlatform.darwin_x64;
-      } else {
-        assert(false);
-      }
-
-      final String engineArtifacts = globals.fs.path.join('bin', 'cache',
-          'artifacts', 'engine');
-      final List<String> paths = <String>[
-        globals.fs.path.join('bin', 'cache', 'dart-sdk', 'bin', 'dart'),
-        globals.fs.path.join('bin', 'cache', 'dart-sdk', 'bin', 'dart.exe'),
-        globals.fs.path.join(engineArtifacts, getNameForHostPlatform(hostPlatform),
-            'frontend_server.dart.snapshot'),
-        globals.fs.path.join(engineArtifacts, 'android-arm-profile',
-            getNameForHostPlatform(hostPlatform), 'gen_snapshot'),
-        globals.fs.path.join(engineArtifacts, 'ios-profile', 'gen_snapshot'),
-        globals.fs.path.join(engineArtifacts, 'common', 'flutter_patched_sdk',
-            'platform_strong.dill'),
-        globals.fs.path.join('lib', 'foo.dart'),
-        globals.fs.path.join('lib', 'bar.dart'),
-        globals.fs.path.join('lib', 'fizz'),
-        globals.fs.path.join('packages', 'flutter_tools', 'lib', 'src', 'build_system', 'targets', 'dart.dart'),
-        globals.fs.path.join('packages', 'flutter_tools', 'lib', 'src', 'build_system', 'targets', 'ios.dart'),
-      ];
-      for (final String path in paths) {
-        globals.fs.file(path).createSync(recursive: true);
-      }
+      iosEnvironment.buildDir.createSync(recursive: true);
     }, overrides: <Type, Generator>{
-      KernelCompilerFactory: () => FakeKernelCompilerFactory(),
-      GenSnapshot: () => FakeGenSnapshot(),
+      Platform: () => FakePlatform(operatingSystem: 'macos', environment: <String, String>{}),
```

Comment:
Its a bit contrived, but I've fixed the platform and filesystem for these tests. They are essentially unit tests anyways, and this cuts out of noise of having `path.join()` absolutely everywhere